src/main/java/strategy/handlers/BalanceHandler.java

```diff
@@ -0,0 +1,12 @@
+package strategy.handlers;
+
+import db.Storage;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class BalanceHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Storage.fruitStorage.put(transaction.getFruit(), transaction.getQuantity());
+    }

```

Comment:
The `BalanceHandler` class directly modifies the `fruitStorage` from the `Storage` class, which violates the encapsulation principle. Consider providing a method in the `Storage` class to handle the updating of the storage to encapsulate the logic within the `Storage` class.